src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1637,7 +1637,7 @@ internal static object CopyDateTimeOffset(object obj)
 
         internal static void SerializeType(object obj, BinaryTokenStreamWriter stream, Type expected)
         {
-            stream.Write(((Type)obj).OrleansTypeName());
+            stream.Write(((Type)obj).OrleansTypeKeyString());
```

Comment:
This PR just makes these two consistent. The types still need to be 'registered' in order for serialization to succeed. Ideally, we would remove that requirement.